Add Multi-Cell Selection Styling sections (#1346)
IgniteUI/igniteui-docfx: en/components/grids_templates/selection.md

```scss
// in component.scss
@import '~igniteui-angular/lib/core/styles/themes/index';
``` 

#### Define colors

Once done, we can make use of the [`igx-contrast-color`]({environment:sassApiUrl}/index.html#function-igx-contrast-color) and [`igx-color`]({environment:sassApiUrl}/index.html#function-igx-color) functions. With them, we define the colors we would like to use for our selection range:
```scss
    $text-color: igx-contrast-color($default-palette, 'primary', 900);
    $background-color: igx-color($default-palette, "primary", 900);
    $border-yellow: #f2c43c;
```

#### Create custom theme
 
Next we create a new theme that extends the [`igx-grid-theme`]({environment:sassApiUrl}/index.html#function-igx-grid-theme) passing our `text-color`, `background-color` and `border-yellow` variables as `$cell-selected-text-color`, `$cell-selected-background` and `$cell-active-border-color`, respectively:

```scss
$custom-grid-theme: igx-grid-theme(
    $cell-selected-text-color: $text-color,
    $cell-active-border-color: $border-yellow,
    $cell-selected-background: $background-color
);
```

#### Apply theme

> Afterwards, all we need to do is include the mixin in our component's style (this could also be done in the `app.root`), so that our @@igSelector uses the newly created theme instead of the default one:

Reviewer: ```suggestion
Afterwards, all we need to do is include the mixin in our component's style (could also be in the app styles), so that our @@igSelector uses the newly created theme instead of the default one:
```

**Note:** Cell selection will trigger [`onSelection`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#onselection) and not [`onRowSelectionChange`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#onrowselectionchange).

### Styling

The theme engine exposes properties that allows us to style the range of selected cells.

#### Import theme

To get started with styling the selection, we need to import the `index` file, where all the theme functions and component mixins live:

```scss
// in component.scss
@import '~igniteui-angular/lib/core/styles/themes/index';
``` 

#### Define colors

Once done, we can make use of the [`igx-contrast-color`]({environment:sassApiUrl}/index.html#function-igx-contrast-color) and [`igx-color`]({environment:sassApiUrl}/index.html#function-igx-color) functions. With them, we define the colors we would like to use for our selection range:
```scss
    $text-color: igx-contrast-color($default-palette, 'primary', 900);
    $background-color: igx-color($default-palette, "primary", 900);
    $border-yellow: #f2c43c;
```

#### Create custom theme
 

> Next we create need to create a new theme that extends the [`igx-grid-theme`]({environment:sassApiUrl}/index.html#function-igx-grid-theme) passing our `text-color`, `background-color` and `border-yellow` variables as `$cell-selected-text-color`, `$cell-selected-background` and `$cell-active-border-color`, respectively:

Reviewer: "Next we **create** need to **create**"

```scss

> $custom-grid-theme: igx-grid-paginator-theme(

Reviewer: paginator theme?

    $cell-selected-text-color: $text-color,
    $cell-active-border-color: $border-yellow,
    $cell-selected-background: $background-color
);
```

#### Consume theme

> Afterwards, all we need to do is have our @@igSelector consume the newly created theme instead of the default one:

Reviewer: The grid (`@@igSelector`) does the consuming of sorts, but what you are doing here is [including the theme mixin](https://sass-lang.com/documentation/at-rules/mixin) in the [app] component styles to override the default theme for the grid inside.

Reviewer: Applies to the title as well. "Apply" would be more suitable.

```scss
    :host { 
        ::ng-deep {
            @include igx-grid($custom-grid-theme);
        }
    }
```

 >[!NOTE]

>  >If the component is using an [`Emulated`](../themes/component-themes.md#view-encapsulation) ViewEncapsulation, it is necessary to `penetrate` this encapsulation using `::ng-deep`:

Reviewer: Make the whole phrase "Emulated ViewEncapsulation" the link. Also keep the back tick (`` ` ``) for code-related references.

> We scope the style under `:host` selector so as not to affect any other grids we might have in our application.

Reviewer: ```suggestion
We scope the style under `:host` selector so as to not affect any other grids we might have in our application.
```

Reviewer: Ugh, this might require a bit more explanation as well.
The `:host` pseudo-selector will only work in Emulated encapsulation or in native shadow DOM. If you go with `None` encapsulation (which I'm sure some do) the host will do exactly.. nothing.
⚠️ I.e. this explanation and the snippet above should clarify they apply only when using encapsulation.

> As a result of the styling, the selected grid cells are highlighted with a more distinctive color:

Reviewer: ```suggestion
With the custom theme applied, the selected grid cells are highlighted with our selected colors:
```
Or something like this...

#### Create custom theme
 
Next we create a new theme that extends the [`igx-grid-theme`]({environment:sassApiUrl}/index.html#function-igx-grid-theme) passing our `text-color`, `background-color` and `border-yellow` variables as `$cell-selected-text-color`, `$cell-selected-background` and `$cell-active-border-color`, respectively:

```scss
$custom-grid-theme: igx-grid-theme(
    $cell-selected-text-color: $text-color,
    $cell-active-border-color: $border-yellow,
    $cell-selected-background: $background-color
);
```

#### Apply theme

Afterwards, all we need to do is include the mixin in our component's style (this could also be done in the `app.root`), so that our @@igSelector uses the newly created theme instead of the default one:

```scss
    @include igx-grid($custom-grid-theme);
```

 >[!NOTE]
 >If the component is using an [`Emulated ViewEncapsulation`](../themes/component-themes.md#view-encapsulation), it is necessary to penetrate this encapsulation using `::ng-deep`:

 ```scss
    :host { 
        ::ng-deep {
            @include igx-grid($custom-grid-theme);
        }
    }
```
 >[!NOTE]

>  > With `Emulated` or `ShadowDОМ` view encapsulation, we scope the style under `:host` selector so as not to affect any other grids we might have in our application.

Reviewer: Okay, not a fan of the two notes.
`:host` scoping is only really needed for `Emulated` and thus should probably be explained in the note above with it.
Basically, `::ng-deep` disables the emulated encapsulation (adding attrs) for its nested styles and Shadow DOM natively does the scoping already.
